### PR TITLE
Adds a client page for Node.js with ioredis

### PR DESCRIPTION
**Pull request by simonprickett**
Adds a page describing how to get up and running with ioredis.

### CODE REVIEW

File: content/rs/references/client_references/client_ioredis.md [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+alwaysopen: false
+categories: ["RS"]
+---
+In order to use Redis with Node.js, you need to install a Node.js Redis client.  In the following sections, we demonstrate the use of [ioredis](https://github.com/luin/ioredis), a community recommended Redis client for Node.js with support for Promises built in.  The other community recommended client for Node.js developers is [node_redis]({{< relref "./client_nodejs" >}}).
+
+Additional Node.js clients for Redis can be found under the [Node.js section](https://redis.io/clients#Node.js) of the [Redis Clients page](https://redis.io/clients).
+
+## Installing ioredis
+
+Complete instructions for installing ioredis can be found in the [README file](https://github.com/luin/ioredis/blob/master/README.md). To install ioredis, issue the following command:
```
Comment (bmansheim) [inline content/rs/references/client_references/client_ioredis.md:14]: s/issue the following command/run/

In general we use the term "run" and assume that the reader knows to run the command that follows the semi-colon.

File: content/rs/references/client_references/client_ioredis.md [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+    npm install ioredis 
+
+## Connecting to Redis
+
+The following code creates a connection to Redis:
```
Comment (bmansheim) [inline content/rs/references/client_references/client_ioredis.md:20]: s/The following/This example/

File: content/rs/references/client_references/client_ioredis.md [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+
+    const Redis = require('ioredis');
+    const redis = new Redis({
+      host: 'hostname',
+      port: 6379,
+      password: 'password'
+    });
+
+Make sure to replace the following values with those for your Redis instance:
```
Comment (bmansheim) [inline content/rs/references/client_references/client_ioredis.md:29]: s/following values/values in the example/

File: content/rs/references/client_references/client_ioredis.md [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+
+- `'hostname'` should be the name of the host your database runs on
```
Comment (bmansheim) [inline content/rs/references/client_references/client_ioredis.md:31]: s/should be/is/

We try not to use modal verbs except for can.

Comment (bmansheim) [reply content/rs/references/client_references/client_ioredis.md:31]: Descriptions in a list use "-" to separate between the term and the description and end with a period if it is a complete sentence.

File: content/rs/references/client_references/client_ioredis.md [hunk 5]
```diff
@@ -0,0 +1,110 @@
+---
+Title: Using Redis with Node.js (ioredis)
+description:
+weight:
+alwaysopen: false
+categories: ["RS"]
+---
+In order to use Redis with Node.js, you need to install a Node.js Redis client.  In the following sections, we demonstrate the use of [ioredis](https://github.com/luin/ioredis), a community recommended Redis client for Node.js with support for Promises built in.  The other community recommended client for Node.js developers is [node_redis]({{< relref "./client_nodejs" >}}).
```
Comment (bmansheim) [inline content/rs/references/client_references/client_ioredis.md:8]: We try put each sentence or sentence segment on it's own line.

File: content/rs/references/client_references/client_ioredis.md [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+        host: 'hostname',
+        port: 6379,
+        tls: {
+            key: fs.readFileSync('path_to_keyfile', encoding='ascii'),
+            cert: fs.readFileSync('path_to_certfile', encoding='ascii'),
+            ca: [ fs.readFileSync('path_to_ca_certfile', encoding='ascii') ]
+        }
+    });
+    
+    ...
```
Comment (bmansheim) [inline content/rs/references/client_references/client_ioredis.md:52]: Are the ... just to show the continuation of the code?

File: content/rs/references/client_references/client_ioredis.md [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+
+- `'hostname'` should be the name of the host your database runs on
+- `6379` is the default Redis port, change this if necessary
+- `'password'` should be the password you use to access Redis.  If a password is not required, remove this line.  Remember to always store passwords outside of your code, for example in environment variables
+
+## Connecting to Redis with TLS 
+
+The following example demonstrates how to configure ioredis to make a connection to Redis using TLS:
+
+    const Redis = require('ioredis');
```
Comment (bmansheim) [inline content/rs/references/client_references/client_ioredis.md:39]: We use ``` to designate code blocks.

File: content/rs/references/client_references/client_ioredis.md (hunk 5 above)
Comment (bmansheim) [reply content/rs/references/client_references/client_ioredis.md:8]: We use single spaces between sentences but if you put the new sentence on a new line then you don't have to put in the space.

File: content/rs/references/client_references/client_ioredis.md [hunk 8]
```diff
@@ -0,0 +1,110 @@
...
+**Note:** ioredis uses the Redis `INFO` command while creating a new connection. Your ACL user must have permission to run that command, otherwise connection attempts will fail with a `NOPERM` error message:
+
+    [ioredis] Unhandled error event: ReplyError: 
+    NOPERM this user has no permissions to run the 'info' command or its 
+    subcommand
+    ...
+
+## Reading and writing data with ioredis
+
+Once connected to Redis, you can start reading and writing data. The following code snippet reads the value stored at the key `foo`, and prints it:
```
Comment (bmansheim) [inline content/rs/references/client_references/client_ioredis.md:79]: We relate to "once" as in how many times and "after" as in which order.

**Review by bmansheim — COMMENTED** [review]
@simonprickett The article looks great! I committed some changes to match our general writing style, but they are in no way a criticism of your writing style.

Look forward to the next contribution!

File: content/rs/references/client_references/client_ioredis.md (hunk 6 above)
Comment (simonprickett) [reply content/rs/references/client_references/client_ioredis.md]: Yes

**Comment by simonprickett** [comment]
@bmansheim thanks for the review and guidance - I did one further commit to change a "should be" for "is".

**Review by bmansheim — APPROVED** [review]

File: content/rs/references/client_references/client_ioredis.md [hunk 9]
```diff
@@ -0,0 +1,143 @@
...
+Complete instructions for installing ioredis can be found in the [README file](https://github.com/luin/ioredis/blob/master/README.md).
+To install ioredis, run:
+
+    npm install ioredis 
+
+## Connecting to Redis
+
+This example code creates a connection to Redis:
+
+    ```sh
```
Comment (gkorland) [inline content/rs/references/client_references/client_ioredis.md:26]: why?     ```sh  

and not ```javascript?

File: content/rs/references/client_references/client_ioredis.md [hunk 10]
```diff
@@ -0,0 +1,143 @@
...
+        const Redis = require('ioredis');
+        const redis = new Redis({
+        host: 'hostname',
```
Comment (gkorland) [inline content/rs/references/client_references/client_ioredis.md:29]: ident the code

**Review by gkorland — CHANGES_REQUESTED** [review]

File: content/rs/references/client_references/client_ioredis.md (hunk 9 above)
Comment (simonprickett) [reply content/rs/references/client_references/client_ioredis.md]: Fixed, thanks.

File: content/rs/references/client_references/client_ioredis.md (hunk 10 above)
Comment (simonprickett) [reply content/rs/references/client_references/client_ioredis.md]: Fixed, including in a couple other places where I've standardized indents, and removed other `sh` blocks making them `javascript` instead.

**Comment by simonprickett** [comment]
@gkorland addressed your comments, thanks.

**Comment by bmansheim** [comment]
@gkorland @banker Any objection to merging this?

**Comment by simonprickett** [comment]
Anything else required here @banker @bmansheim - I think this is good to go?

**Pull request merged**